opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/ScriptUtils.java

```diff
@@ -22,7 +21,8 @@ public class ScriptUtils {
    * Limitation: assume inner field name is always "keyword".
    */
   public static String convertTextToKeyword(String fieldName, ExprType fieldType) {
-    if (fieldType == OPENSEARCH_TEXT_KEYWORD) {
+    if (fieldType instanceof OpenSearchTextType
+        && ((OpenSearchTextType) fieldType).getFields().size() > 0) {
```

Comment:
Should the OpenSearchDataType has interface to resoved field?

Reply:
A type could have multiple fields, which could be accessed by `getFields` method.

Reply:
Fixed in d2f4ed5e9.